### ISSUE TITLE
sequences.snipped doesn't handle negative indices

### ISSUE THREAD

**Issue by erg** [factor/factor#3104]
```factor
-1 { 1 2 3 } <removed> >array
! i kind of expect the whole array, { 1 2 3 }
```

**Comment by mrjbq7** [comment]
From Discord

> [10:12 AM] knottio: While trying out sequences.snipped for AOC, I noticed the public constructors don't do bounds checks, allowing out-of-bounds access, e.g.:

```factor
IN: scratchpad 10 <iota> 3 head-slice dup >array . 10 swap <removed> >array .
{ 0 1 2 }
{ 0 1 2 3 4 5 6 7 8 9 }
```

**Comment by mrjbq7** [comment]
Also related is positive out of bounds indices

**Comment by mrjbq7** [comment]
Fixed in 09fc16a1cba15292f28b0644e75bc0ac5c8fce70
